### PR TITLE
use core instead of std for wasm sanity checks

### LINKED ISSUE #274: sanity_checks_for_wasm depends on std
Trying to build `secp256k1-sys` currently fails in a `no_std` context because of a `std` dependency in [types.rs](https://github.com/rust-bitcoin/rust-secp256k1/blob/master/secp256k1-sys/src/types.rs#L70).

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

Closes #274

### CODE REVIEW

File: secp256k1-sys/src/types.rs [hunk 1]
```diff
@@ -43,13 +45,12 @@ impl AlignedType {
...
 
-
 #[cfg(test)]
 mod tests {
     extern crate libc;
-    use std::os::raw;
-    use std::mem;
     use std::any::TypeId;
+    use std::mem;
+    use std::os::raw;
```
Comment (dr-orlovsky) [inline secp256k1-sys/src/types.rs:53]: Seems this also should be replaced with `core`

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]
NB: Have not checked that PR covers all `std` uses in the lib

File: secp256k1-sys/src/types.rs (hunk 1 above)
Comment (elichai) [reply secp256k1-sys/src/types.rs:53]: well these aren't in core :)

**Review by elichai — APPROVED** [review]
My mistake, sorry.

I wonder if we can arrange a no-std wasm test here

File: secp256k1-sys/src/types.rs (hunk 1 above)
Comment (dr-orlovsky) [reply secp256k1-sys/src/types.rs:53]: Below in line 66 we have `core::mem`

**Comment by gregdhill** [comment]
@elichai can we merge this and work on a no-std wasm test in a separate PR?

**Comment by elichai** [comment]
@gregdhill yeah no need to add a test here.
Just waiting for a second review

File: secp256k1-sys/src/types.rs (hunk 1 above)
Comment (gregdhill) [reply secp256k1-sys/src/types.rs:53]: `any::TypeId` and `mem` are in `core` but `os::raw` isn't

**Review by apoelstra — APPROVED** [review]
ack c86808fac70a3adf7ce180ae62151cf841914893

**Pull request merged**